tokio-console/src/warnings.rs

```diff
@@ -46,6 +50,31 @@ pub trait Warn<T>: Debug {
     fn summary(&self) -> &str;
 }
 
+/// A result for a linter check
+pub(crate) enum Lint<T> {
+    /// No warning applies to the entity
+    Ok,
+
+    /// The cloned instance of `Self` should be held by the entity that
+    /// generated the warning, so that it can be formatted. Holding the clone of
+    /// `Self` will increment the warning count for that entity.
+    Warning(Linter<T>),
+
+    /// The lint should be rechecked as the conditions to allow for checking are
+    /// not satisfied yet
+    Recheck,
+}
+
+/// A result for a warning check
+pub enum Warning {
+    /// Set `true` if the warning applies or `false` otherwise
+    Check(bool),
```

Comment:
I feel it might be simpler to have two separate enum variants, instead of keeping a bool in a single variant.

```suggestion
    /// No warning for this entity.
    Ok,
    
    /// A warning has been detected for this entity.
    Warn,
```